### PR TITLE
Add Support For Downloading Videos From YouTube

### PR DESCRIPTION
**Pull request by robwebset**
Hi, Thanks for your work on the YouTube Addon!

I have put together some code that adds support for downloading videos from YouTube by bring up the context menu on a given video and selecting download.

I have made use of the youtube-dl module that is already available in the official repo (I didn't see the point in adding in duplicate code in your addon, when it was pretty easy to call the module already available).

I hope you don't mind me submitting the pull request - I have tried to follow your existing coding style so that it fits with what you already have.

Thanks, Rob

### CODE REVIEW

File: resources/lib/youtube/helper/yt_download.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from resources.lib.youtube.youtube_exceptions import YouTubeException
+from resources.lib.youtube.helper import utils
+
+# Method to download a given Video to the disk, will prompt the user for where to store it
+def download_video(provider, context, video_id):
+    try:
+#        download_location = xbmcgui.Dialog().browse(3, "Save Video Location", 'files')
+#        context.log_debug("Download location set to %s" % download_location)
+
+        url = "http://www.youtube.com/watch?v=%s" % video_id
```
Comment (bromix) [inline resources/lib/youtube/helper/yt_download.py]: This won't work with private (your) or age restricted videos (only with public videos)...you can use ```get_video_streams``` of the client and you get all URLs for the video stream.

**Comment by uNiversaI** [comment]
@robwebset if you willing to help maintain a separate youtbube addon with the fixes bromix does and this patch I wil be more than happy to help you.

**Comment by robwebset** [comment]
@uNiversaI I'm afraid I don't really use the YouTube addon much myself - I did the pull request so that users of VideoExtras would be able to download "Extras" from youtube and store them with their Movies.  I just don't really have the time to maintain anything else - and don't really want to re-badge the work that @bromix has already done - I'm not sure that is the right way to go (Seems a bit wrong to take the work of someone else and rebadge it) - I believe from conversations with @bromix that this is still on the todo list.

**Comment by uNiversaI** [comment]
@robwebset Its opensource or no? Nothing wrong with doing that in opensource thats what it is,
In any case he closed https://github.com/bromix/plugin.video.youtube/issues/37 which seems to indicate little interest due to not enough requests.

Ive many people bugging me and atm we all using youtube-dl or firefox addons or similar its a much needed feature for many reasons including those you mention but ok, Ill do it for personal use then as the youtube-dl interface and way to achieve this is rather cumbersome.

**Pull request closed without merging**